pkg/config/config.go

```diff
@@ -219,9 +272,36 @@ func (c *MicroshiftConfig) ReadFromConfigFile(configFile string) error {
 	if err != nil {
 		return fmt.Errorf("reading config file %q: %v", configFile, err)
 	}
+	var config Config
+	if err := yaml.Unmarshal(contents, &config); err != nil {
+		return fmt.Errorf("decoding config file %s: %v", configFile, err)
+	}
 
-	if err := yaml.Unmarshal(contents, c); err != nil {
-		return fmt.Errorf("decoding config file %q: %v", configFile, err)
+	// Wire new Config type to existing MicroshiftConfig
+	c.LogVLevel = config.GetVerbosity()
+	if config.NodeName != "" {
+		c.NodeName = config.NodeName
+	}
+	if config.NodeIP != "" {
+		c.NodeIP = config.NodeIP
+	}
+	if config.URL != "" {
+		c.Cluster.URL = config.URL
+	}
+	if len(config.Network.ClusterNetwork) != 0 {
+		c.Cluster.ClusterCIDR = config.Network.ClusterNetwork[0].CIDR
+	}
+	if len(config.Network.ServiceNetwork) != 0 {
+		c.Cluster.ServiceCIDR = config.Network.ServiceNetwork[0]
+	}
+	if config.Network.ServiceNodePortRange != "" {
+		c.Cluster.ServiceNodePortRange = config.Network.ServiceNodePortRange
+	}
+	if config.DNS.BaseDomain != "" {
+		c.Cluster.Domain = config.DNS.BaseDomain
+	}
```

Comment:
```suggestion
	if config.DNS.ClusterDomain != "" {
		c.Cluster.Domain = config.DNS.ClusterDomain
	}
```

This change is quite indicative of what I was explaining above.